qiskit_experiments/library/randomized_benchmarking/mirror_rb_analysis.py

```diff
@@ -0,0 +1,674 @@
+        )
+
+    def _create_analysis_results(
+        self,
+        fit_data: curve.FitData,
+        quality: str,
+        **metadata,
+    ) -> List[AnalysisResultData]:
+        """Create analysis results for important fit parameters.
+
+        Args:
+            fit_data: Fit outcome.
+            quality: Quality of fit outcome.
+
+        Returns:
+            List of analysis result data.
+        """
+
+        outcomes = super()._create_analysis_results(fit_data, quality, **metadata)
+        num_qubits = len(self._physical_qubits)
+
+        # nrb is calculated for both EPC and EI per the equations in the docstring
+        ei_nrb = 4**self._num_qubits
+        ei_scale = (ei_nrb - 1) / ei_nrb
+        epc_nrb = 2**self._num_qubits
+        epc_scale = (epc_nrb - 1) / epc_nrb
+
+        alpha = fit_data.ufloat_params["alpha"]
+
+        # Calculate EPC and EI per the equations in the docstring
+        epc = epc_scale * (1 - alpha)
+        ei = ei_scale * (1 - alpha)
+
+        outcomes.append(
+            AnalysisResultData(
+                name="EPC", value=epc, chisq=fit_data.reduced_chisq, quality=quality, extra=metadata
+            )
+        )
+        outcomes.append(
+            AnalysisResultData(
+                name="EI", value=ei, chisq=fit_data.reduced_chisq, quality=quality, extra=metadata
+            )
+        )
+
+        # Correction for 1Q depolarizing channel if EPGs are provided
+        if self.options.epg_1_qubit and num_qubits == 2:
+            epc = _exclude_1q_error(
+                epc=epc,
+                qubits=self._physical_qubits,
+                gate_counts_per_clifford=self._gate_counts_per_clifford,
+                extra_analyses=self.options.epg_1_qubit,
+            )
+            outcomes.append(
+                AnalysisResultData(
+                    name="EPC_corrected",
+                    value=epc,
+                    chisq=fit_data.reduced_chisq,
+                    quality=quality,
+                    extra=metadata,
+                )
+            )
+
+        # Calculate EPG
+        if self._gate_counts_per_clifford is not None and self.options.gate_error_ratio:
+            epg_dict = _calculate_epg(
+                epc=epc,
+                qubits=self._physical_qubits,
+                gate_error_ratio=self.options.gate_error_ratio,
+                gate_counts_per_clifford=self._gate_counts_per_clifford,
+            )
+            if epg_dict:
+                for gate, epg_val in epg_dict.items():
+                    outcomes.append(
+                        AnalysisResultData(
+                            name=f"EPG_{gate}",
+                            value=epg_val,
+                            chisq=fit_data.reduced_chisq,
+                            quality=quality,
+                            extra=metadata,
+                        )
+                    )
+
+        return outcomes
+
+    def _run_data_processing(
```

Comment:
This is not correct inheritance of `CurveAnalysis`. If you just need to compute `_compute_polarizations_and_probabilities` you should implement this function as [a node](https://github.com/Qiskit/qiskit-experiments/blob/9fe27bf65a4a939c43db49a4e205a713568377ae/qiskit_experiments/data_processing/data_action.py#L25) of `DataProcessor`, and define a custom data processor with it in the default analysis option. This will avoid lots of boilerplate code here. There is no automatic method to propagate a base class change to subclass, and this implementation drastically increases maintenance overhead.

Reply:
Thanks, this is gone now.

Reply:
Awesome. It's really clean now